common/roschannels/monitorchannel.hpp

```diff
@@ -22,12 +30,99 @@ namespace ROSChannels {
                 BasePub<message_type>(node, "object_" + std::to_string(id) + "/" + topicName, qos) {}
         };
 
+        class AnchorPub : public BasePub<message_type> {
+        public:
+            AnchorPub(rclcpp::Node& node, const rclcpp::QoS& qos = defaultQoS) :
+                BasePub<message_type>(node, "object_anchor/" + topicName, qos) {}
+        };
+
         class Sub : public BaseSub<message_type> {
         public:
             const uint32_t objectId;
             Sub(rclcpp::Node& node, const uint32_t id, std::function<void(const message_type::SharedPtr)> callback, const rclcpp::QoS& qos = defaultQoS) : 
                 objectId(id),
                 BaseSub<message_type>(node, "object_" + std::to_string(id) + "/" + topicName, callback, qos) {}
         };
+
+        class AnchorSub : public BaseSub<message_type> {
+        public:
+            AnchorSub(rclcpp::Node& node, std::function<void(const message_type::SharedPtr)> callback, const rclcpp::QoS& qos = defaultQoS) : 
+                BaseSub<message_type>(node, "object_anchor/" + topicName, callback, qos) {}
+        };
+
+        static message_type fromISOMonr(const uint32_t id, const ObjectMonitorType& indata) {
+            atos_interfaces::msg::Monitor outdata;
+            auto txid = id;
+            auto stamp = rclcpp::Time(indata.timestamp.tv_sec, indata.timestamp.tv_usec*1000);
+            
+            // Set stamp for all subtypes
+            outdata.atos_header.header.stamp = stamp;
+            outdata.pose.header.stamp = stamp;
+            outdata.velocity.header.stamp = stamp;
+            outdata.acceleration.header.stamp = stamp;
+
+            // Set frame ids
+            outdata.atos_header.header.frame_id = "map"; // TODO
+            outdata.pose.header.frame_id = "map"; // TODO
+            outdata.velocity.header.frame_id = "map"; // TODO vehicle local
+            outdata.acceleration.header.frame_id = "map"; // TODO vehicle local
+
+            outdata.atos_header.object_id = txid;
+            outdata.object_state.state = indata.state;
+            if (indata.position.isPositionValid) {
```

Comment:
I was just wondering about the isValid check on the Z coordinate a few lines down on 75. I am assuming we are still allowing z to be ignored. I am just thinking if this could be an issue for drones. But then again, I since this is the monitor message I assume this is due to iso22133.

Reply:
added isvalid checks for x and y too